src/mongocxx/instance.cpp

```diff
@@ -14,28 +14,63 @@
 
 #include <mongocxx/instance.hpp>
 
+#include <mongoc.h>
+#include <utility>
+
 #include <bsoncxx/stdx/make_unique.hpp>
 
-#include <mongoc.h>
+#include <mongocxx/private/libmongoc.hpp>
+#include <mongocxx/logger.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
 
 namespace {
-void log_handler(mongoc_log_level_t, const char *, const char *, void *) {
+extern "C" void null_log_handler(::mongoc_log_level_t, const char *, const char *, void *) {
```

Comment:
Maybe move this down next to user_log_handler so it is easier to see the relationship